File: Algorithm-CPP/27_N의표현법.cpp
#include <stdio.h>
#include <vector>
using namespace std;

bool isPrime(int x) {
    int cnt = 0;

    for(int i = 2; i <= x; i++) {
        if(x % i == 0) cnt++;
    }

    if(cnt > 1) return false;
    else return true;
}

int main() {
    int n;
    vector<int> primes(1001);

    scanf("%d", &n);

    for(int i = 2; i <= n; i++) {
        int temp = i;
        for(int j = 2; temp > 1; ) {
            if(temp % j == 0) {
                primes[j]++;
                temp /= j;
            } else j++;
        }
    }

    printf("%d! = ", n);
    for(int i = 2; i <= n; i++) {
        if(isPrime(i)) {
            printf("%d ", primes[i]);
        }
    }

    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/27_N의표현법.swift
//
//  27_N의표현법.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/08/07.
//

import Foundation

func isPrime3(_ num: Int) -> Bool {
    var cnt = 0

    for i in 2...num {
        if num % i == 0 {
            cnt += 1
        }
    }

    return cnt == 1
}

func solution27() {
    let n = Int(readLine()!)!
    var primes = Array(repeating: 0, count: 1001)

    for i in  2...n {
        var temp = i
        var j = 2

        while true {
            if temp % j == 0 {
                primes[j] += 1
                temp /= j
            } else {
                j += 1
            }

            if temp == 1 {
                break
            }
        }
    }

    print("\(n)! =", terminator: " ")
    for i in 2...n {
        if isPrime3(i) {
            print("\(primes[i])", terminator: " ")
        }
    }

}
